src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -529,10 +537,43 @@ namespace winrt::TerminalApp::implementation
     // - <none>
     // Return Value:
     // - <none>
-    void TerminalPage::_CompleteInitialization()
+    winrt::fire_and_forget TerminalPage::_CompleteInitialization()
     {
         _startupState = StartupState::Initialized;
-        _InitializedHandlers(*this, nullptr);
+
+        // GH#632 - It's possible that the user tried to create the terminal
+        // with only one tab, with only an elevated profile. If that happens,
+        // we'll create _another_ process to host the elevated version of that
+        // profile. This can happen from the jumplist, or if the default profile
+        // is `elevate:true`, or from the commandline.
+        //
+        // However, we need to make sure to close this window in that scenario.
+        // Since there aren't any _tabs_ in this window, we won't ever get a
+        // closed event. So do it manually.
+        auto weakThis{ get_weak() };
+        if (_tabs.Size() == 0)
+        {
+            // This is MENTAL. If we exit right away after spawning the elevated
+            // WT, then ShellExecute might not successfully complete the
+            // elevation. What's even more, the Terminal will mysteriously crash
+            // somewhere in XAML land.
+            //
+            // So I'm introducing a 5s delay here for the Shell to complete the
+            // execution, and _then_ close the window.
+            //
+            // TODO! There's no way this is the right answer, right?
+            co_await winrt::resume_background();
+            if (auto page{ weakThis.get() })
+            {
+                Sleep(5000);
```

Comment:
Isn't there literally a `co_await <duration>` instead of juggling to a background thread and then back to the foreground?